### PR TITLE
Added proto representation for GPU & related metrics

### PR DESCRIPTION
**Pull request by JacksonM8**

### CODE REVIEW

File: re/re_common/proto/aggregationmessage/aggregationmessage.proto [hunk 1]
```diff
@@ -97,6 +97,15 @@ message NetworkUtilisationResponse {
     repeated NodeNetworkEvents node_network_events = 1;
 }
 
+message GPUMetricRequest {
+    repeated google.protobuf.Timestamp time_interval = 1;
+    repeated string node_ids = 2;
+    uint32 experiment_run_id = 3;
+    repeated string node_hostnames = 4;
+}
+message GPUUtilisationResponse {
```
Comment (cathlynaston) [inline re/re_common/proto/aggregationmessage/aggregationmessage.proto:106]: Rename to GPUMetricResponse?

File: re/re_common/proto/aggregationmessage/aggregationmessage.proto [hunk 2]
```diff
@@ -197,6 +206,21 @@ message NetworkUtilisationEvent {
...
+    repeated GPUDeviceMetrics per_device_metrics = 2;
+}
+message GPUDeviceMetrics {
+    int32 gpu_device_index = 1;
+    repeated GPUMetricSample samples = 2;
+}
+message GPUMetricSample {
+    google.protobuf.Timestamp time = 1;
+    double gpu_utilisation = 2;
+    int64 mem_utilisation = 3;
```
Comment (cathlynaston) [inline re/re_common/proto/aggregationmessage/aggregationmessage.proto:220]: Was this meant to be a double like the MemoryUtilisationEvent?

File: re/re_common/proto/aggregationmessage/aggregationmessage.proto (hunk 1 above)
Comment (JacksonM8) [reply re/re_common/proto/aggregationmessage/aggregationmessage.proto:106]: Good catch, will do

File: re/re_common/proto/aggregationmessage/aggregationmessage.proto (hunk 2 above)
Comment (JacksonM8) [reply re/re_common/proto/aggregationmessage/aggregationmessage.proto:220]: I'll be receiving the memory usage as the raw number of MiB, so if I rename it to include the units this should hopefully be a little clearer.

**Review by cathlynaston — APPROVED** [review]

**Pull request merged**